fedimint-api/src/db/mod.rs

```diff
@@ -46,6 +46,24 @@ pub trait Database: Send + Sync {
     fn raw_find_by_prefix(&self, key_prefix: &[u8]) -> PrefixIter<'_>;
 
     fn raw_apply_batch(&self, batch: DbBatch) -> Result<()>;
+
+    fn new_transaction<'a>(&'a self) -> Box<dyn DbTransaction<'a> + 'a>;
```

Comment:
I would advise to wrap `Box<dyn DbTransaction<'a> + 'a>` into a neat newtype that `Derefer`s to &dyn DBTransaction` to avoid typing all that all the time. We're going to be using that pattern all over the modularization framework as well. Example: https://github.com/fedimint/fedimint/pull/546/files#diff-de9b10a6ed1a7ca9958bd8722504beb2edd17137c8ded2c261379e3fee1f9357R26